SnackOverflow/src/main/java/ro/utcn/sd/vasi/SnackOverflow/controller/QuestionHandler.java

```diff
@@ -0,0 +1,139 @@
+        questionManagementService.listQuestions().forEach(x->{
+            UserData user = userManagementService.getUserData(x.getAuthorId()).orElse(null);
+            print(x.toStringNoText() + " author: " + user.getUsername() + " score: " + user.getScore());});
+    }
+
+    private void handleAddQuestion() {
+        print("Enter question title");
+        String title = scanner.nextLine().trim();
+
+        print("Enter question text");
+        String text = scanner.nextLine().trim();
+
+        print("Existing tags: " + questionManagementService.listAllTags());
+        print("New tags will be automatically added");
+        print("Please input the tags, comma separated");
+        String tagStr = scanner.nextLine().trim();
+        Set<Tag> tags = new TreeSet<>(Arrays.asList(tagStr.toLowerCase().split(",")).stream().map(x->new Tag(x)).collect(Collectors.toList()));
+        tags.removeIf(x->x.getName().equals(""));
+        questionManagementService.addQuestion(currentUser.getId(),title,text,tags);
+        print("Question successfully added");
+    }
+
+    private void handleFilterByTag() {
+        print("Existing tags: " + questionManagementService.listAllTags());
+        print("Please input the tags, comma separated");
+        String tagStr = scanner.nextLine().trim();
+        Set<Tag> tags = new TreeSet<>(Arrays.asList(tagStr.toLowerCase().split(",")).stream().map(x->new Tag(x)).collect(Collectors.toList()));
+
+        List<Question> questions = questionManagementService.filterQuestionsByTag(tags);
+
+        questions.forEach(x->{
+            UserData user = userManagementService.getUserData(x.getAuthorId()).orElse(null);
+            print(x.toStringNoText() + " author: " + user.getUsername() + " score: " + user.getScore());});
+
+        if(questions.isEmpty()) print("No questions");
+    }
+
+    private void handleFilterByTitle() {
+        print("Please input the title");
+        String title = scanner.nextLine().trim();
+
+        List<Question> questions = questionManagementService.filterQuestionsByTitle(title);
+        questions.forEach(x->{
+            UserData user = userManagementService.getUserData(x.getAuthorId()).orElse(null);
+            print(x.toStringNoText() + " author: " + user.getUsername() + " score: " + user.getScore());});
+
+        if(questions.isEmpty()) print("No questions");
+    }
+
+    private void handleSeeQuestionAnswers() {
+        print("Enter question id");
+        int id = scanner.nextInt();
+        scanner.nextLine();
+
+        Question question = questionManagementService.getQuestion(id);
+        UserData userQ = userManagementService.getUserData(question.getAuthorId()).orElse(null);
+        print(question.toString() + " author: " + userQ.getUsername() + " score: " + userQ.getScore());
+
+        List<Answer> ans = answerManagementService.listAnswersForQuestion(id);
+        if(ans.isEmpty()) print("No answers");
```

Comment:
Writing if-then-else statements like this is a little hard to read -- always use curly brackets, even for one single line of code.